Formula/ibazel.rb

```diff
@@ -0,0 +1,22 @@
+class Ibazel < Formula
+  desc "Tools for building Bazel targets when source files change"
+  homepage "https://github.com/bazelbuild/bazel-watcher"
+  url "https://github.com/bazelbuild/bazel-watcher/archive/refs/tags/v0.16.2.tar.gz"
+  sha256 "a927520e7ab3a1fb749043e543c57bb211666cd627d053fbe0e8245730beee75"
+  license "Apache-2.0"
+
+  depends_on "bazel" => :build
+  depends_on xcode: :build
+
+  def install
+    system "bazel", "build", "--config=release", "--workspace_status_command", "echo STABLE_GIT_VERSION #{version}", "//ibazel"
+    bin.install "bazel-bin/ibazel/ibazel_/ibazel"
+  end
+
+  test do
+    # Since ibazel loops in most cases the quickest check of validity
+    # I can think of is to get the version output which happens when
+    # invoked without any arguments.
+    system bin/"ibazel"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@SMillerDev Ibazel by design continues running until the user terminates it. Do you have any suggestions for a suitable test?

Reply:
OK, I can test building a sample app.